.github/workflows/lint-and-test.yml

```diff
@@ -0,0 +1,49 @@
+name: 'Linting and Testing'
+
+on: pull_request
+
+concurrency:
+  group: lint-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  run-tests:
+    name: 'Run Tests'
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
```

Comment:
We don't have a use case that requires entire history. So, I guess we can get rid of this.

Reply:
Good point, I can at least remove to see if everything works fine.

Reply:
Everything works fine with `fetch-depth` removed!